### PR TITLE
[SPARK-30715][K8S] Bump fabric8 to 4.7.1

### PR DESCRIPTION
**Pull request by onursatici**
### What changes were proposed in this pull request?
Bump fabric8 kubernetes-client to 4.7.1

### Why are the changes needed?
New fabric8 version brings support for Kubernetes 1.17 clusters.
Full release notes:
- https://github.com/fabric8io/kubernetes-client/releases/tag/v4.7.0
- https://github.com/fabric8io/kubernetes-client/releases/tag/v4.7.1

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Existing unit and integration tests cover creation of K8S objects. Adjusted them to work with the new fabric8 version

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
ok to test

File: core/src/test/resources/fairscheduler-with-valid-data.xml [hunk 1]
```diff
@@ -32,4 +32,4 @@
         <weight>3</weight>
         <schedulingMode>FAIR</schedulingMode>
     </pool>
-</allocations>
```
Comment (dongjoon-hyun) [inline core/src/test/resources/fairscheduler-with-valid-data.xml]: Please revert this file change.

**Comment by dongjoon-hyun** [comment]
Hi, @onursatici . Is this for improvement or for a bug fix?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
@onursatici . Ur, could you run `K8s` integration test on this PR locally first?
```
[INFO] Compiling 19 Scala sources to /home/jenkins/workspace/SparkPullRequestBuilder-K8s/resource-managers/kubernetes/integration-tests/target/test-classes ...
[ERROR] [Error] /home/jenkins/workspace/SparkPullRequestBuilder-K8s/resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/DepsTestsSuite.scala:58: ')' expected but string literal found.
[ERROR] [Error] /home/jenkins/workspace/SparkPullRequestBuilder-K8s/resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/DepsTestsSuite.scala:59: ';' expected but ')' found.
```

**Comment by dongjoon-hyun** [comment]
@onursatici . Since you are active contributor in K8s area, I want to recommend to keep all sections of the PR template. Especially, `### Why are the changes needed?` section. You can add the following release note at that section simply.
- https://github.com/fabric8io/kubernetes-client/releases/tag/v4.7.0
- https://github.com/fabric8io/kubernetes-client/releases/tag/v4.7.1

In short, please recover the deleted two sections.

**Comment by dongjoon-hyun** [comment]
It seems that there exist 3 failures.
```
KubernetesSuite:
- Run SparkPi with no resources *** FAILED ***
  The code passed to eventually never returned normally. Attempted 70 times over 2.0014648788833336 minutes. Last failure message: false was not true. (KubernetesSuite.scala:315)
- Run SparkPi with a very long application name. *** FAILED ***
  The code passed to eventually never returned normally. Attempted 70 times over 2.0008292973333335 minutes. Last failure message: false was not true. (KubernetesSuite.scala:315)
...
- Launcher client dependencies *** FAILED ***
  The code passed to eventually never returned normally. Attempted 130 times over 4.00124932035 minutes. Last failure message: false was not true. (KubernetesSuite.scala:315)
```

**Comment by onursatici** [comment]
Thanks for having a look @dongjoon-hyun, will fix the integrations tests and will let you know

**Comment by onursatici** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Oh, it seems that two failures are gone. The remaining one is known to be a flaky test.
```
- Launcher client dependencies *** FAILED ***
  The code passed to eventually never returned normally. Attempted 130 times over 4.0007343233 minutes. Last failure message: false was not true. (KubernetesSuite.scala:315)
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

I verified this locally. The flaky failure is irrelevant to this PR.
```
KubernetesSuite:
- Run SparkPi with no resources
- Run SparkPi with a very long application name.
- Use SparkLauncher.NO_RESOURCE
- Run SparkPi with a master URL without a scheme.
- Run SparkPi with an argument.
- Run SparkPi with custom labels, annotations, and environment variables.
- All pods have the same service account by default
- Run extraJVMOptions check on driver
- Run SparkRemoteFileTest using a remote data file
- Run SparkPi with env and mount secrets.
- Run PySpark on simple pi.py example
- Run PySpark with Python2 to test a pyfiles example
- Run PySpark with Python3 to test a pyfiles example
- Run PySpark with memory customization
- Run in client mode.
- Start pod creation from template
- PVs with local storage
- Launcher client dependencies
- Run SparkR on simple dataframe.R example
Run completed in 13 minutes, 30 seconds.
Total number of tests run: 19
Suites: completed 2, aborted 0
Tests: succeeded 19, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```

**Comment by dongjoon-hyun** [comment]
Thank you, @onursatici .

**Comment by dongjoon-hyun** [comment]
I'll backport this to `branch-3.0` to address the recent JDK8 issue on K8s 1.17.x.

**Pull request closed without merging**